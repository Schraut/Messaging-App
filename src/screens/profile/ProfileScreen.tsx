import * as React from "react";
import { Text, View, StyleSheet, SafeAreaView, Image } from "react-native";
import { Theme, useTheme } from "../../../resources/theme/theme";
import ProfileHeader from "./ProfileHeader";
import SettingsButton from "../../components/buttons/SettingsButton";
import Icon from "../../components/icon/Icon";

interface ProfileScreenProps {
  navigation: any; //we may not need this. Depends on if we take it out of bottomtabs
  firstName: string;
  lastName: string;
  username: string;
  profileImage: string;
}

export default function ProfileScreen({
  navigation,
  firstName,
  lastName,
  username,
  profileImage,
}: ProfileScreenProps) {
  const theme = useTheme();
  const styles = stylesHandler(theme);

  return (
    <SafeAreaView style={styles.container}>
      <ProfileHeader navigation={navigation} />
      <View style={styles.accountInfo}>
        <Image source={{ uri: profileImage }} style={styles.profileImage} />
        <Text style={styles.nameText}>
          {firstName} {lastName}
        </Text>
        <Text style={styles.usernameText}>{username}</Text>
      </View>
      <View style={styles.settingsButtonContainer}>
        <SettingsButton
          onPress={() => console.log("Goes to Account Settings")}
          text="Account Settings"
          icon={
            <Icon
              iconName="account-circle"
              size={24}
              iconType="MaterialCommunityIcons"
              color={theme.colors.inverseBackground}
            />
          }
        />
        <SettingsButton
          onPress={() => console.log("Goes to Edit Profile")}
          text="Edit Profile"
          icon={
            <Icon
              iconName="edit"
              size={18}
              iconType="FontAwesome5"
              color={theme.colors.inverseBackground}
            />
          }
        />
        <SettingsButton
          onPress={() => console.log("Goes to Language")}
          text="Language"
          icon={
            <Icon
              iconName="language"
              size={24}
              iconType="Ionicons"
              color={theme.colors.inverseBackground}
            />
          }
        />
      </View>
      <View style={{ height: theme.spacing(6) }} />
      <View style={styles.settingsButtonContainer}>
        <SettingsButton
          onPress={() => console.log("Goes to Privacy Policy")}
          text="Privacy Policy"
          icon={
            <Icon
              iconName="account-circle"
              size={24}
              iconType="MaterialCommunityIcons"
              color={theme.colors.inverseBackground}
            />
          }
        />
        <SettingsButton
          onPress={() => console.log("Goes to Security")}
          text="Security"
          icon={
            <Icon
              iconName="lock-check"
              size={24}
              iconType="MaterialCommunityIcons"
              color={theme.colors.inverseBackground}
            />
          }
        />
      </View>
    </SafeAreaView>
  );
}

const stylesHandler = (theme: Theme) =>
  StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: theme.colors.background,
      alignItems: "center",
    },
    accountInfo: {
      width: "100%",
      justifyContent: "center",
      alignItems: "center",
      paddingVertical: theme.spacing(5),
    },
    profileImage: {
      width: theme.spacing(25),
      height: theme.spacing(25),
      borderRadius: theme.spacing(25) / 2, //to get 50
    },
    nameText: {
      color: theme.colors.inverseBackground,
      fontFamily: theme.font.header.fontFamily,
      fontSize: theme.font.header.fontSize,
    },
    usernameText: {
      color: theme.colors.inverseBackground,
      fontFamily: theme.font.textPrimary.fontFamily,
      fontSize: theme.font.textPrimary.fontSize,
    },
    text: {
      color: theme.colors.inverseBackground,
      fontFamily: theme.font.textPrimary.fontFamily,
      fontSize: theme.font.textPrimary.fontSize,
      marginTop: theme.spacing(2),
    },
    settingsButtonContainer: {
      backgroundColor: theme.colors.background,
      width: theme.spacing(85),
      padding: theme.spacing(4),
      borderRadius: theme.spacing(2),
      elevation: 2, //come back to compare android and ios later
      shadowColor: theme.colors.staticGrey, //shadow for ios
      shadowOffset: {
        width: 0,
        height: 4,
      },
      shadowOpacity: 0.6,
      justifyContent: "center",
      alignItems: "flex-start",
    },
  });
