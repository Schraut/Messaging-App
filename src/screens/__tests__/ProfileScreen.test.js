import React from "react";
import renderer from "react-test-renderer";
import ThemeContent from "../../../resources/theme/theme";
import ProfileScreen from "../profile/ProfileScreen";

describe("Default ProfileScreen", () => {
  it("renders in light mode", () => {
    const tree = renderer
      .create(
        <ThemeContent>
          <ProfileScreen />
        </ThemeContent>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders in dark mode", () => {
    const tree = renderer
      .create(
        <ThemeContent isDarkMode={true}>
          <ProfileScreen />
        </ThemeContent>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
