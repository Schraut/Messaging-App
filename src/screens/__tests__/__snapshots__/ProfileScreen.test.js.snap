// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default ProfileScreen renders in dark mode 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "space-between",
        "paddingHorizontal": 4,
        "width": 750,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#000000",
          "borderRadius": 40,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 40,
        }
      }
    >
      <View>
        <View />
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#000000",
          "borderRadius": 40,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 40,
        }
      }
    >
      <View>
        <View />
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingVertical": 20,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "uri": undefined,
        }
      }
      style={
        {
          "borderRadius": 50,
          "height": 100,
          "width": 100,
        }
      }
    />
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 24,
        }
      }
    >
       
    </Text>
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 12,
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "flex-start",
        "backgroundColor": "#000000",
        "borderRadius": 8,
        "elevation": 2,
        "justifyContent": "center",
        "padding": 16,
        "shadowColor": "#7E7E7E",
        "shadowOffset": {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.6,
        "width": 340,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#000000",
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 28,
            }
          }
        >
          <View>
            <View />
          </View>
        </View>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
            }
          }
        >
          Account Settings
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#000000",
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 28,
            }
          }
        >
          <View>
            <View />
          </View>
        </View>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
            }
          }
        >
          Edit Profile
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#000000",
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 28,
            }
          }
        >
          <View>
            <View />
          </View>
        </View>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
            }
          }
        >
          Language
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "height": 24,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "flex-start",
        "backgroundColor": "#000000",
        "borderRadius": 8,
        "elevation": 2,
        "justifyContent": "center",
        "padding": 16,
        "shadowColor": "#7E7E7E",
        "shadowOffset": {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.6,
        "width": 340,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#000000",
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 28,
            }
          }
        >
          <View>
            <View />
          </View>
        </View>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
            }
          }
        >
          Privacy Policy
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#000000",
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 28,
            }
          }
        >
          <View>
            <View />
          </View>
        </View>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
            }
          }
        >
          Security
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Default ProfileScreen renders in light mode 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "space-between",
        "paddingHorizontal": 4,
        "width": 750,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 40,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 40,
        }
      }
    >
      <View>
        <View />
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 40,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 40,
        }
      }
    >
      <View>
        <View />
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingVertical": 20,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "uri": undefined,
        }
      }
      style={
        {
          "borderRadius": 50,
          "height": 100,
          "width": 100,
        }
      }
    />
    <Text
      style={
        {
          "color": "#000000",
          "fontFamily": "System",
          "fontSize": 24,
        }
      }
    >
       
    </Text>
    <Text
      style={
        {
          "color": "#000000",
          "fontFamily": "System",
          "fontSize": 12,
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "flex-start",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 8,
        "elevation": 2,
        "justifyContent": "center",
        "padding": 16,
        "shadowColor": "#7E7E7E",
        "shadowOffset": {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.6,
        "width": 340,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 28,
            }
          }
        >
          <View>
            <View />
          </View>
        </View>
        <Text
          style={
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
            }
          }
        >
          Account Settings
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 28,
            }
          }
        >
          <View>
            <View />
          </View>
        </View>
        <Text
          style={
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
            }
          }
        >
          Edit Profile
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 28,
            }
          }
        >
          <View>
            <View />
          </View>
        </View>
        <Text
          style={
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
            }
          }
        >
          Language
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "height": 24,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "flex-start",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 8,
        "elevation": 2,
        "justifyContent": "center",
        "padding": 16,
        "shadowColor": "#7E7E7E",
        "shadowOffset": {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.6,
        "width": 340,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 28,
            }
          }
        >
          <View>
            <View />
          </View>
        </View>
        <Text
          style={
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
            }
          }
        >
          Privacy Policy
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 28,
            }
          }
        >
          <View>
            <View />
          </View>
        </View>
        <Text
          style={
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
            }
          }
        >
          Security
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
